Integrations/integration-Preempt.yml

```diff
@@ -0,0 +1,411 @@
rTuAchXyi3WJdC0Yz2eQQW0w/vKRcrlcVR737oMWkP/Lt1tVx9y8YNwLq27Knsrc5sWA4pnCsUN1newLoKm3K64q6P7WxdB1/txqVr/MXejAYDmMegDgNbgaj5kxV3aKP3HugT6zg/CRT2+QANa7TnlbOXmOKp9a10G7dMY4J6ov6Wnat3CS591urRtK6BQtnXHl461l62LoG+0V/i6VK272yXuYd0F8NLn9DHoA4AmMegDgNZY1boA0MBtO/PP/VJ+MusSQAG4q7weUNwX0o9al0FnNQa617n4Qbiy1oqyllKy7AUUQD/leGVT6yJoivv7yKAPWbCycrl1CQDIOwZ9ANAaXNGHLHBXMTxoXQJ95wfhz7VsZt0DyLlRXnoL42oc1f5sXQb29D54TMtj+hsbaD1M2crjBxVAMzbW8bS4jq0XrYugz55U3lYWsK0BfDfoAwA0iUEfADSpVK0vrGVe6x6A3JRUyqOsS6ApR3vpL+UB9J77+zdMGRJHtVetyyB79L6ItGznB+EJWo9StvYY+AF94a6MPcbjx0m5pT1Don3sDXp5pHUXdL153Hcqek/+zboIAOQZgz4AaB5X8yErhlkXQN/5QfgzLQOtewA55G7R6Z41dBwDPvSE3ievadm+MfD7jZde4cePLIDe2UTH0C90PL1kXQR9dr3HoA/ZsLLCoA8AmsCgDwCat5J1AUD+llTKz1iXQFO4mg/ovfuUo+Ko9oJ1EeSP3jeva9nWD8Kq1iHK+saVgDwZc1Xf5tZF0DfaO7xSqtbd5+eS1l3Q9VZWLrMuAQB5xqAPAJrHoA9ZcKN1AfSdH4SLatnCugeQI88qB8dR7QnrIsi/xi09N9Df4l9pPUP5pXElIC825aq+3HNXxDPog7WVrQsAQN4x6AOAJpSq9R9pmc+6ByA3WBdAU9yzoriaD5i0f3jpbcauj6NaYl0GxaL31JN+EC6nl9t66RV+8xtXArLOnbu4OxLwY6X8coO+k730Ck3Aytw8pw8AmsOgDwCaw9V8yII/a1PEc6lyyg/ChbRsad0DyLjhyunKyXFU+8q6DIqrMUC+Tn+bb9NaUQ5TprZtBWTaZjpeftJ49iVyRnuIt0vV+h/1cnnrLuh67qp6Bn0A0EcM+gCgOQz6kAU3WxdAUw5RJrMuAWTYPcoBcVR707oIuofeb19rOdYPwqFaz1I2Mq4EZJW7qs8NxHe1LoI+c48AYNAHaysqV1mXAIC8YtAHAM1h0IcsuMW6APrGD8I5texo3QPIKHebzsFxVLvTugi6l95/b2nZWH+v3aDvPI9btgPjs52OkWN0vLxnXQR9cquX/qCB28jD0q+sCwBAnjHoA4A+KlXrc2lZyLoHut7zSaX8hnUJ9NkBylTWJYCMGeWlAxX3pfEX1mUAR+/Fu/wgfFgvj1X299hLA2ObQjnQS+9SgJzRXuKf2ts+5TFoga2F9T6cU+/Hf1kXAYA8YnMCAH0XWhcAPK7myy0/CPtr2cu6B5Axf1F2jaPan6yLAOPS+/JLLYfo77e7zd2Vyi+MKwFZsoeOjSE6Tj61LoI+cXsKBn2w5t6Dt1qXAIA8YtAHAH23gnUBwOP5fHnmhnz9rUsAGTFSOUU5MY5q31iXASZG79G6H4Rlvfy1coQyuXElIAumVwYpQ6yLoE9u87h9J+wx6AOAPmLQBwB9xxV9sPZCUin/3boEes8Pwim99NZvADzvVWWHOKo9a10E6Cm9X7/V8hv9Pf+t1muVRY0rAVkwWMfEmTo+vrYugt7RnuK9UrX+tMceF7a4qhQA+ohBHwD0gTZB02hZ0roHut7t1gXQZ9soc1mXAIwlygVKJY5qX1mXAfpC793n/CBc2kuvSN1PKRlXAizNrmynXGZdBH3i9hYM+mBp8VK1Pm1SKf/XuggA5A2DPgDom2U8btMEe3dYF0CfcTUfut2/lZ3jqHavdRGgWY2rl/b3g/A+rUOVOYwrAZbcsXC5jovEugh6ze0tzrAuga7mvqd2t8Z+3LoIAOQNgz4A6Bt+6QhrryeV8l+sS6D3/CBcRcvi1j0AQw956a0637cuArSS3tMP6G+8+/vuhn1rWfcBjCymrK48aF0EvaO9xVulav3PermEdRd0teU9Bn0A0GsM+gCgb5a3LoCux9V8+cXVfOhWo5RjlFPiqDbaugzQDnpvf+AH4bp6eahyoseeG93Jnesw6Msnt8dg0AdLK1gXAIA8YtMBAH2zrHUBdD0GfTnkB+GCWjaw7gEY+EDZOo5qj1oXAdqtMciu6m/+n7QOU+Y0rgR02jp6/y+iY+F16yLoNbfHOM66BLracqVqvZRUytz+FwB6gUEfAPSSTjrdF/WzWfdAV3O3u3vGugT6ZD+ln3UJoMNqyhZxVPundRGgk/Sef8wPwqX08iZlRes+QAe5c53Byr7WRdA7SaX8kva7b+nlj6y7oGu571oWUv5mXQQA8oRBHwD0Hlfzwdo9/MIxf/wgnEHLztY9gA67VNkvjmrfWhcBLLhnUerv/2p6ebYyyLoP0EE76r1/lI6Bz6yL/B97dx4l21nWDXtX8mUeCAECAUKFOQwWEFoUCzAyKYMvMsgMCsrkiIo28CIKitKKggwqyieKEOZJUECIhKEQsBJCAWEIAQpICIRAJjKck5x67213Vk5OTndXn9O97131XNda99qFf7h+Sap27/389sCGvadaLmohy09Wij6ADVH0AWycoo9s78kOwB55Qszh2SGgIXWxVxd8/5AdBLLF72B7bH692+t/JravjDkgORI04dCYX4r5m+wgbNi/VYo+ct0t5l+zQwDMEkUfwMYp+sh0ScxJ2SHYI0/PDgANOS/mYePR4KPZQaBN4jfxmm6v/8Vq+R1YHgNPCZ4e3/mXx3ffkyhmS/33u74T84jsIBTrx7MDAMwaRR/ABnSWhvUV2HfOzkHRTposLlySHYKN6fb69buZ7pidAxrwlZgHj0cDj1uC3YjfxiD+Jtw9Pr435rjsPLDF6u/4C
```

Comment:
According to our context conventions
`activities.EndpointHostName` should be `Preempt.Activities.EndpointHostName` 
...
`endpoint.Hostname` => `Endpoint.Hostname` (Endpoint we already have in the context)
![image](https://user-images.githubusercontent.com/7270217/37582792-09edf7ca-2b68-11e8-8c1e-13d55d88215c.png)

Reply:
So
```python 
"EntryContext": {"activities": pretty_results, "cursor": cursor}
```
should become 
```python 
"EntryContext": {"Preempt": {"Activities": pretty_results}, "cursor": cursor}
```
?

Reply:
It should be 
```
"EntryContext": {
     "Preempt.Activities(val.ID==obj.ID)": pretty_results,
     "Preempt.Cursor": cursor
}
```

The `ID` can be replaced with the unique identifier field of single activity

Reply:
I don't get the part of (val.ID==obj.ID). What is that for?

Reply:
This used to prevent duplicate objects in the context. This means iterate over all the activities ,if in the array exists activity with the same ID then update that activity , otherwise add new activity to the list.
For that to happen activity or any other object need to have some unique identifier field. 
According to your code there is no such identifier.  
```
def prettyfy_result(res):
            return {
                "EventType": res["eventType"],
                "AuthenticationType": res["authenticationType"],
                "Timestamp": res["timestamp"],
                "EndpointHostName": res["endpointEntity"]["hostName"]
            }
```
So you can do just
```

"EntryContext": {
     "Preempt.Activities": pretty_results,
     "Preempt.Cursor": cursor
}
```

Reply:
Ah, ok. Actually the activities are guaranteed to come back unique from the API call.